content/vscode/remote.md

```diff
@@ -0,0 +1,134 @@
+
+# dotnetthailand ขอเสนอ [Remote Development extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack)
+
+สิ่งที่ต้องเตรียมก็มี (เครื่องเปล่าเปิดซิงก็ทำตามได้ หรือมีอะไรลงไปแล้วก็ข้ามๆไป)
+
+- ลง [Docker](https://docs.docker.com/get-docker/) ก่อนเลย
+- แน่นอนลง [VS Code](https://code.visualstudio.com/download)
+- เสร็จแล้วลง [Remote Development extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack)
+- แค่นั้นแหละจบ
+
+# ก่อนไปลุยทำความเข้าใจมันนิดว่ามันทำงานยังไง
+
+![alt text][architecture]
+
+ref: https://code.visualstudio.com/docs/remote/containers
+
+แบบสั้นๆ หลักการคือจะใช้เครื่องเราทำหน้าที่เก็บ code เฉยๆ และใช้ VS Code ที่ลงบนเครื่องทำหน้าที่แค่แสดงผล แต่กระบวนการ build, run อะไรก็ตามจะไปทำงานใน container โดยผ่านการ mount volume ของ code ที่เราทำงาน
+
+ภาษาที่ Support:
+
+- dotnet core 2, 3, 5
+- nodejs 12, 14, 16
+- PHP
+- Golang
+- Java
+- Python
+- Ruby
+- Rust
+- และอีกเพียบไปลองไล่ดูกันเอา
+
+และไม่ใช่แค่สำหรับ dev อย่างเดียวแต่ยังมี container อย่าง azure cli, Docker in Docker, Kubernetes ที่ทำให้เราไม่ต้องมาลงของพวกนี้ในเครื่องอีกต่อไป ใช้ผ่าน container แล้วเขียน script โลด
+
+# แล้ววิธีใช้ทำยังไง
+
+ขอยกตัวอย่างจากการที่ผู้เขียนทำการ folk project web ตัวนี้เอามาเขียนเนื้อหานี่แหละ
+
+1. แน่นอน clone code ลงมา (หรือจะสร้าง folder เปล่าก็ได้)
+2. เปิด VS Code ปกติเราจะสั่งเปิด folder อย่าเพิ่งจิ้มให้กด cmd + P บน Mac (ผู้เขียนลืม Windows ไปแล้ว) แล้วเลือก *Remote-Containers: Open Folder in Container...*
+
+![alt text][open]
+
+3. ตัว web นี้ based on Node ก็แน่นอนเลือก Node ซิ และเลือกเป็น v 14 ด้วยนะ v 16 ผู้เขียนลองแล้ว build ไม่ผ่าน :P
+
+![alt text][selectNode]
+![alt text][v14]
+
+4. รอไปซักพักนึงเพราะมันจะ build image ให้ในเครื่องเราแล้วจะ start ขึ้นมาให้แล้วเราจะเจอ folder *.devcontainer* พร้อมอีก 2 files ข้างใน
+
+![alt text][devcontainer]
+
+5. ตัว *devcontainer.json* หน้าตาตอนเปิดครั้งแรกก็จะประมาณนี้
+
+```json
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.177.0/containers/javascript-node
+{
+	"name": "Node.js",
+	"build": {
+		"dockerfile": "Dockerfile",
+		// Update 'VARIANT' to pick a Node version: 12, 14, 16
+		"args": { "VARIANT": "14" }
+	},
+
+	// Set *default* container specific settings.json values on container create.
+	"settings": { 
+		"terminal.integrated.shell.linux": "/bin/bash"
+	},
+
+	// Add the IDs of extensions you want installed when the container is created.
+	"extensions": [
+		"dbaeumer.vscode-eslint"
+	],
+
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
+	// "forwardPorts": [],
+
+	// Use 'postCreateCommand' to run commands after the container is created.
+	// "postCreateCommand": "yarn install",
+
+	// Comment out connect as root instead. More info: https://aka.ms/vscode-remote/containers/non-root.
+	"remoteUser": "node"
```

Comment:
IMHO, the comments in this code block except the first two comments are not much value and some are code comment , could you please remove them. 
I think it will help our code shorter and easy to read.

Reply:
I want to show all the contents which coming from tool itself.

Reply:
Okay